packages/apptive_grid_core/lib/src/model/form/after_submit_action.dart

```diff
@@ -0,0 +1,101 @@
+/// The AfterSubmitAction class represents an action to be taken after a form
+/// is submitted.
+///
+/// Currently only the custom button title is used in the app.
+class AfterSubmitAction {
+  /// Creates a new instance of the AfterSubmitAction class from the given JSON
+  /// data.
+  factory AfterSubmitAction.fromJson(dynamic json) => AfterSubmitAction(
+        type: AfterSubmitActionType.values
+            .firstWhere((type) => type.name == json['action']),
+        buttonTitle: json['buttonTitle'],
+        trigger: json['trigger'] != null
+            ? AfterSubmitActionTrigger.values
+                .firstWhere((trigger) => trigger.name == json['trigger'])
+            : null,
+        delay: json['delay'] != null ? Duration(seconds: json['delay']) : null,
+        targetUrl:
+            json['targetUrl'] != null ? Uri.parse(json['targetUrl']) : null,
+      );
+
+  /// Creates a new instance of the AfterSubmitAction class.
+  const AfterSubmitAction({
+    required this.type,
+    this.buttonTitle,
+    this.trigger,
+    this.delay,
+    this.targetUrl,
+  });
+
+  /// The type of the action to be taken after form submission.
+  final AfterSubmitActionType type;
+
+  /// The title of the button that was clicked to trigger the action, if
+  /// applicable.
+  final String? buttonTitle;
+
+  /// The trigger that caused the action to be taken, if applicable.
+  final AfterSubmitActionTrigger? trigger;
+
+  /// The delay (in seconds) before the action is taken, if applicable.
```

Comment:
The `(in seconds)` is now not needed anymore

Reply:
done